packages/sdk/ios/Sources/RNTPrimer.swift

```diff
@@ -361,8 +362,9 @@ extension RNTPrimer: PrimerDelegate {
 
             DispatchQueue.main.async {
                 do {
-                    let checkoutPaymentmethodData = try JSONEncoder().encode(data)
-                    let checkoutPaymentmethodJson = try JSONSerialization.jsonObject(with: checkoutPaymentmethodData, options: .allowFragments)
+                    let checkoutPaymentmethodJson = try data
+                        .toPrimerCheckoutPaymentMethodDataRN()
+                        .toJsonObject()
                     self.sendEvent(withName: PrimerEvents.onBeforePaymentCreate.stringValue, body: checkoutPaymentmethodJson)
```

Comment:
Yes, you are right - my mistake.
Threw me since the code is now matching the interface - but of course merchants could have code set up around the incorrect type.... 
+1 for static typing